server/src/main/java/org/opensearch/index/store/remote/filecache/FileCacheCleaner.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.remote.filecache;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.apache.logging.log4j.util.Supplier;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.env.NodeEnvironment;
+import org.opensearch.index.IndexModule;
+import org.opensearch.index.shard.IndexEventListener;
+import org.opensearch.index.shard.ShardId;
+import org.opensearch.index.shard.ShardStateMetadata;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.opensearch.index.store.remote.directory.RemoteSnapshotDirectoryFactory.LOCAL_STORE_LOCATION;
+
+/**
+ * IndexEventListener to clean up file cache when the index is deleted
```

Comment:
It's probably worth adding an explanation of why this exists. I'm thinking something like:

> IndexEventListener to clean up file cache when the index is deleted. The cached entries will be eligible for eviction when the shard is deleted, but this listener deterministically removes entries from memory and from disk at the time of shard deletion as opposed to waiting for the cache to need to perform eviction.

What do you think?